### PR TITLE
Feature/gmail data

### PR DESCRIPTION
**Pull request by willarion**
messed up with commits history, forgot to make new branch, therefore changes after previous PR are here

### CODE REVIEW

File: src/components/Post/Post.tsx [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+  id: string;
 }
 
-function getProperDate(dateString: string): string {
+const getProperDate = (dateString?: string): string => {
+  if (!dateString) {
+    return '';
+  }
   const date = new Date(dateString);
+  if (date.getDate() !== date.getDate()) {
```
Comment (bondarchukse) [inline src/components/Post/Post.tsx]: what are you doing here?
you can separated function to check date validity
````
const isValidDate = (date: Date): boolean => {
 return date instanceof Date && isFinite(date);
}
````

File: src/components/Post/Post.tsx [hunk 2]
```diff
@@ -33,10 +42,17 @@ export const Post: React.FC<PostsProps> = ({
...
-          todo change a to link and path to proper one
-          <a href="index.html" className="media-heading" title="Item title">
+          {/*todo change a to link and path to proper one*/}
+          <Link
+            to={{
+              pathname: '/message',
+              state: id,
+            }}
+            className="media-heading"
+            title="Item title"
```
Comment (bondarchukse) [inline src/components/Post/Post.tsx]: you can set message title instead of hardcode

File: src/pages/MessagePage/MessagePage.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import './MessagePage.css';
+
+export const MessagePage: React.FC = ({}) => {
+  const location = useLocation();
+  const id = location.state;
+  const token = useAuthToken();
+
+  const [message, setMessage] = React.useState<UserMessage | null>(null);
+
+  React.useEffect(() => {
```
Comment (bondarchukse) [inline src/pages/MessagePage/MessagePage.tsx]: move this logic to the custom hook!

File: src/components/Post/Post.tsx [hunk 4]
```diff
@@ -1,22 +1,31 @@
 import React from 'react';
+import { Link } from 'react-router-dom';
 
 interface PostsProps {
```
Comment (bondarchukse) [inline src/components/Post/Post.tsx]: this anme little bit confusing

Comment (willarion) [reply src/components/Post/Post.tsx]: what is anme?

**Pull request merged**